File: rust_bindings/java/src/test/java/app/grafica/twittertext/TestHitHighlighter.java
package app.grafica.twittertext;

import static org.junit.Assert.*;
import org.junit.Test;

import java.io.IOException;
import java.util.List;
import java.util.Map;

public class TestHitHighlighter {
  static {
    TwitterTextJNILib.load();
  }

  @Test
  public void testConstructor() {
    HitHighlighter highlighter = new HitHighlighter();
    assertNotNull(highlighter);
  }

  @Test
  public void testConstructorParam() {
    HitHighlighter highlighter = new HitHighlighter("hmm");
    assertNotNull(highlighter);
  }

  private static final String KEY_HIGHLIGHT_HITS = "hits";

  private static Hits makeHits(List<List<Integer>> ints) {
    Hits hits = new Hits();
    for (List<Integer> list : ints) {
      Hit hit = new Hit();
      hit.setStart(list.get(0));
      hit.setEnd(list.get(1));
      hits.add(hit);
    }
    return hits;
  }

  @Test
  public void testYaml() throws IOException {
    HitHighlighter highlighter = new HitHighlighter();
    String path = "rust/conformance/tests/hit_highlighting.yml"; 
    List<Map> tests = Yaml.loadConformanceData(path, "plain_text");
    assertNotNull(tests);
    assert(tests.size() > 0);
    for (Map test : tests) {
      Hits hits = makeHits((List<List<Integer>>) test.get(KEY_HIGHLIGHT_HITS));
      assertEquals(test.get("expected"), highlighter.highlight(test.get("text").toString(), hits));
    }

    tests = Yaml.loadConformanceData(path, "with_links");
    assertNotNull(tests);
    assert(tests.size() > 0);
    for (Map test : tests) {
      Hits hits = makeHits((List<List<Integer>>) test.get(KEY_HIGHLIGHT_HITS));
      assertEquals(test.get("expected"), highlighter.highlight(test.get("text").toString(), hits));
    }
  }
}

File: rust/python-bindings/test/test_hithighlighter.py
import os
import pytest
import sys
import twitter_text
import yaml

def test_ctor():
    hit_highlighter = twitter_text.HitHighlighter()
    assert hit_highlighter is not None

def test_ctor_with_string():
    hit_highlighter = twitter_text.HitHighlighter("hmm")
    assert hit_highlighter is not None

def array_to_hits(arr):
    hits = twitter_text.Hits()
    for hit in arr:
        h = twitter_text.Hit()
        h.start = hit[0]
        h.end = hit[1]
        hits.append(h)

    return hits

def test_yaml():
    hit_highlighter = twitter_text.HitHighlighter()
    with open (r"rust/conformance/tests/hit_highlighting.yml") as file:
        testmap = yaml.load(file, Loader=yaml.FullLoader)

        assert len(testmap["tests"]["plain_text"]) > 0
        for test in testmap["tests"]["plain_text"]:
            assert test["expected"] == hit_highlighter.highlight(test["text"], array_to_hits(test["hits"]))

        assert len(testmap["tests"]["with_links"]) > 0
        for test in testmap["tests"]["with_links"]:
            assert test["expected"] == hit_highlighter.highlight(test["text"], array_to_hits(test["hits"]))

if __name__ == "__main__":
    raise SystemExit(pytest.main([__file__]))
